apps/docs-snippets/src/guide/predicates/index.test.ts

```diff
@@ -18,7 +18,11 @@ describe(__filename, () => {
     // #import { Predicate, Provider, FUEL_NETWORK_URL };
 
     const provider = await Provider.create(FUEL_NETWORK_URL);
-    const predicate = new Predicate(binary, provider, jsonAbi);
+    const predicate = new Predicate({
+      bytecode: binary,
+      provider,
+      abi: jsonAbi,
+    });
```

Comment:
Would assigning the parameters to a variable and then using the variable in the constructor be more descriptive? I am thinking mainly about the type annotation, but it may not be necessary.

```ts
const predicateParams: Type = { .. }
```

Reply:
I think it might not be a good idea in this case because doing this would require us to pass in a shape for the predicate data which might introduce more confusion: 
<img width="844" alt="image" src="https://github.com/FuelLabs/fuels-ts/assets/39617427/57a15c73-ed1a-432a-8ea9-bc2e9e85cdc2">

Reply:
Or maybe we should always encourage people to pass in the input type 🤔

Reply:
I think it could be kinda confusing because the class automatically infers the type if you do pass in `inputData` like so:

<img width="503" alt="image" src="https://github.com/FuelLabs/fuels-ts/assets/39617427/65147f88-498b-40d0-ba69-b329dd14b005">